mbuild/packing.py

```diff
@@ -313,6 +320,156 @@ def fill_region(compound, n_compounds, region, overlap=0.2,
+        Specify that compounds should not be rotated when filling the sphere,
+        default=False.
+    temp_file : str, default=None
+        File name to write PACKMOL's raw output to.
+
+    Returns
+    -------
+    filled : mb.Compound
+
+    """
+    _check_packmol(PACKMOL)
+
+    arg_count = 2 - [n_compounds, density].count(None)
+    if arg_count != 1:
+        msg = ("Exactly 1 of `n_compounds` and `density` "
+               "must be specified. {} were given.".format(arg_count))
+        raise ValueError(msg)
+
+    if isinstance(sphere, (list, set)):
+        if len(sphere) != 4:
+            msg = ("`sphere` must be a list of len 4")
+    else:
+        msg = ("`sphere` must be a list")
+        raise ValueError(msg)
+
+    if not isinstance(compound, (list, set)):
+        compound = [compound]
+    if n_compounds is not None and not isinstance(n_compounds, (list, set)):
+        n_compounds = [n_compounds]
+    if not isinstance(fix_orientation, (list, set)):
+        fix_orientation = [fix_orientation]*len(compound)
+
+    if compound is not None and n_compounds is not None:
+        if len(compound) != len(n_compounds):
+            msg = ("`compound` and `n_compounds` must be of equal length.")
+            raise ValueError(msg)
+
+    if compound is not None:
+        if len(compound) != len(fix_orientation):
+            msg = ("`compound`, `n_compounds`, and `fix_orientation` must be of equal length.")
+            raise ValueError(msg)
+
+    for coord in sphere[:3]:
+        if coord < sphere[3]:
+            msg = ("`sphere` center coordinates must be greater than radius.")
+            raise ValueError(msg)
+
+    # Apply edge buffer
+    radius = sphere[3] - edge
+
+    if density is not None:
+        if n_compounds is None:
+            if len(compound) == 1:
+                compound_mass = np.sum([a.mass for a in compound[0].to_parmed().atoms])
+                # Conversion from kg/m^3 / amu * nm^3 to dimensionless units
+                n_compounds = [int(density/compound_mass*(4/3*np.pi*radius**3)*.60224)]
+            else:
+                if compound_ratio is None:
+                    msg = ("Determing `n_compounds` from `density` "
+                           "for systems with more than one compound type requires"
+                           "`compound_ratio`")
+                    raise ValueError(msg)
+                if len(compound) != len(compound_ratio):
+                    msg = ("Length of `compound_ratio` must equal length of "
+                           "`compound`")
+                    raise ValueError(msg)
+                prototype_mass = 0
+                for c, r in zip(compound, compound_ratio):
+                    prototype_mass += r * np.sum([a.mass for a in c.to_parmed().atoms])
+                # Conversion from kg/m^3 / amu * nm^3 to dimensionless units
+                n_prototypes = int(density/prototype_mass*(4/3*np.pi*radius**3)*.60224)
```

Comment:
should move this to a variable, we use this twice. will be easier to deal with in the future.

Reply:
the conversion factor i mean